lsp-ui-doc.el

```diff
@@ -1164,25 +1164,18 @@ It is supposed to be called from `lsp-ui--toggle'"
   (interactive)
   (lsp-ui-doc--hide-frame))
 
-(defvar-local lsp-ui-doc--unfocus-frame-timer nil)
 (defun lsp-ui-doc--glance-hide-frame ()
-  "Hook to hide hover information popup for `lsp-ui-doc-glance'."
-  (when (or (overlayp lsp-ui-doc--inline-ov)
```

Comment:
> Note that `lsp-ui-doc--glance-hide-frame` is also simplified to just remove itself from `post-command-hook` and directly call `lsp-ui-doc-hide`.

This is done purposely to make sure that when the doc frame is focused, we can scroll inside the doc frame (the `post-command-hook` is triggered) and when we back out to the document, the `lsp-ui-doc--hide-frame` will be triggered.
The condition here is to ignore the calling to hide frame (and remove this function from `post-command-hook`) when the doc frame is visible

Reply:
I can understand not wanting to auto-close it when the doc frame is focused, but what has that got to do with visibility? That sounds like it could be the cause of the original bug (where the doc window triggered via glance was never auto-hiding).

Reply:
Sorry I misread the condition. The condition here actually is "Don't call and unhook from `post-command-hook` yet when the doc frame is not visible (which can take time to set up)".

That's also the root cause of the original bug, which happens for overlay. Currently, we only check if the overlay is there, not checking its visibility. Thus the `lsp-ui-doc--hide-frame` will be called when the overlay is not even visible yet.
The simplest change is just to add a new condition there.
``` el
  (when (or (and (overlayp lsp-ui-doc--inline-ov) (overlay-buffer lsp-ui-doc--inline-ov))
```

You may want to spin it into a new function mimicking the frame version, `lsp-ui-doc--overlay-visible-p`)

Reply:
No, it turns out it works the first time. The root cause is that the overlay variable isn't being cleared when the overlay itself is deleted. I've updated the PR.

Reply:
I think we purposely don't reset the overlay that way so that a new overlay doesn't have to be created every time the inline doc is hidden.
The condition checking `(overlay-buffer lsp-ui-doc--inline-ov)` returns the buffer that the overlay is in, if it becomes `nil`, means the overlay has been deleted, which is similar to what you do if you clear the overlay. The good thing is that we don't have to create a new overlay